C API broken again

commit 0a7e077ed has broken the C API again replacing all C headers by their C++ versions

```
In file included from /usr/local/include/spatialindex/capi/sidx_api.h:34:
/usr/local/include/spatialindex/capi/sidx_config.h:57:13: fatal error: 'cstdint' file not found
   #include <cstdint>
            ^~~~~~~~~
```

--- a/include/spatialindex/capi/BoundsQuery.h
+++ b/include/spatialindex/capi/BoundsQuery.h
@@ -29,20 +29,20 @@
 #pragma once
 
 #include "sidx_export.h"
-#include <memory>
 
 class SIDX_DLL BoundsQuery : public SpatialIndex::IQueryStrategy
 {
 private:
-    std::unique_ptr< SpatialIndex::Region > m_bounds;
+    SpatialIndex::Region* m_bounds;
     
 public:
 
     BoundsQuery();
+    ~BoundsQuery() { if (m_bounds != 0) delete m_bounds; }
     void getNextEntry(  const SpatialIndex::IEntry& entry, 
                         SpatialIndex::id_type& nextEntry, 
-                        bool& hasNext) override;
+                        bool& hasNext);
     
-    SpatialIndex::Region* GetBounds() const { return m_bounds.get(); }
+    SpatialIndex::Region* GetBounds() const { return m_bounds; }
 };
 

--- a/include/spatialindex/capi/CountVisitor.h
+++ b/include/spatialindex/capi/CountVisitor.h
@@ -33,16 +33,16 @@
 class SIDX_DLL CountVisitor : public SpatialIndex::IVisitor
 {
 private:
-   uint64_t nResults{0};
+   uint64_t nResults;
 
 public:
 
    CountVisitor();
-   ~CountVisitor() override;
+   ~CountVisitor();
 
    uint64_t GetResultCount() const { return nResults; }
 
-   void visitNode(const SpatialIndex::INode& n) override;
-   void visitData(const SpatialIndex::IData& d) override;
-   void visitData(std::vector<const SpatialIndex::IData*>& v) override;
+   void visitNode(const SpatialIndex::INode& n);
+   void visitData(const SpatialIndex::IData& d);
+   void visitData(std::vector<const SpatialIndex::IData*>& v);
 };

--- a/include/spatialindex/capi/CustomStorage.h
+++ b/include/spatialindex/capi/CustomStorage.h
@@ -37,15 +37,21 @@ namespace SpatialIndex
         struct SIDX_DLL CustomStorageManagerCallbacks
         {
             CustomStorageManagerCallbacks() 
-            = default;
+            : context(0)
+            , createCallback(0)
+            , destroyCallback(0)
+            , loadByteArrayCallback(0)
+            , storeByteArrayCallback(0)
+            , deleteByteArrayCallback(0)
+            {}
 
-            void* context{nullptr};
-            void (*createCallback)( const void* context, int* errorCode ){nullptr};
-            void (*destroyCallback)( const void* context, int* errorCode ){nullptr};
+            void* context;
+            void (*createCallback)( const void* context, int* errorCode );
+            void (*destroyCallback)( const void* context, int* errorCode );
 			void (*flushCallback)( const void* context, int* errorCode );
-            void (*loadByteArrayCallback)( const void* context, const id_type page, uint32_t* len, byte** data, int* errorCode ){nullptr};
-            void (*storeByteArrayCallback)( const void* context, id_type* page, const uint32_t len, const byte* const data, int* errorCode ){nullptr};
-            void (*deleteByteArrayCallback)( const void* context, const id_type page, int* errorCode ){nullptr};
+            void (*loadByteArrayCallback)( const void* context, const id_type page, uint32_t* len, byte** data, int* errorCode );
+            void (*storeByteArrayCallback)( const void* context, id_type* page, const uint32_t len, const byte* const data, int* errorCode );
+            void (*deleteByteArrayCallback)( const void* context, const id_type page, int* errorCode );
         };
 
         class SIDX_DLL CustomStorageManager : public SpatialIndex::IStorageManager
@@ -58,12 +64,12 @@ namespace SpatialIndex
 
 	        CustomStorageManager(Tools::PropertySet&);
 
-	        ~CustomStorageManager() override;
+	        virtual ~CustomStorageManager();
 
-			void flush() override;
-	        void loadByteArray(const id_type page, uint32_t& len, byte** data) override;
-	        void storeByteArray(id_type& page, const uint32_t len, const byte* const data) override;
-	        void deleteByteArray(const id_type page) override;
+			virtual void flush();
+	        virtual void loadByteArray(const id_type page, uint32_t& len, byte** data);
+	        virtual void storeByteArray(id_type& page, const uint32_t len, const byte* const data);
+	        virtual void deleteByteArray(const id_type page);
 
         private:
             CustomStorageManagerCallbacks   callbacks;

--- a/include/spatialindex/capi/DataStream.h
+++ b/include/spatialindex/capi/DataStream.h
@@ -34,22 +34,23 @@ class SIDX_DLL DataStream : public SpatialIndex::IDataStream
 {
 public:
     DataStream(int (*readNext)(SpatialIndex::id_type* id, double **pMin, double **pMax, uint32_t *nDimension, const uint8_t **pData, size_t *nDataLength));
+    ~DataStream();
 
-    SpatialIndex::IData* getNext() override;
-    bool hasNext() override;
+    SpatialIndex::IData* getNext();
+    bool hasNext();
 
-    uint32_t size() override;
-    void rewind() override;
+    uint32_t size();
+    void rewind();
 
 protected:
-    std::unique_ptr< SpatialIndex::RTree::Data > m_pNext;
+    SpatialIndex::RTree::Data* m_pNext;
     SpatialIndex::id_type m_id;
 
 private:
     int (*iterfunct)(SpatialIndex::id_type *id, double **pMin, double **pMax, uint32_t *nDimension, const uint8_t **pData, size_t *nDataLength);
 
     bool readData();
-    bool m_bDoneReading { false };
+    bool m_bDoneReading;
 
 };
 

--- a/include/spatialindex/capi/IdVisitor.h
+++ b/include/spatialindex/capi/IdVisitor.h
@@ -34,17 +34,17 @@ class SIDX_DLL IdVisitor : public SpatialIndex::IVisitor
 {
 private:
     std::vector<uint64_t> m_vector;
-    uint64_t nResults{0};
+    uint64_t nResults;
 
 public:
 
     IdVisitor();
-    ~IdVisitor() override;
+    ~IdVisitor();
 
     uint64_t GetResultCount() const { return nResults; }
     std::vector<uint64_t>& GetResults()  { return m_vector; }
     
-    void visitNode(const SpatialIndex::INode& n) override;
-    void visitData(const SpatialIndex::IData& d) override;
-    void visitData(std::vector<const SpatialIndex::IData*>& v) override;
+    void visitNode(const SpatialIndex::INode& n);
+    void visitData(const SpatialIndex::IData& d);
+    void visitData(std::vector<const SpatialIndex::IData*>& v);
 };

--- a/include/spatialindex/capi/LeafQuery.h
+++ b/include/spatialindex/capi/LeafQuery.h
@@ -39,10 +39,11 @@ private:
 	std::vector<LeafQueryResult> m_results;
 public:
 
-    LeafQuery() = default;
+	LeafQuery();
+	~LeafQuery() { }
 	void getNextEntry(	const SpatialIndex::IEntry& entry, 
 						SpatialIndex::id_type& nextEntry, 
-						bool& hasNext) override;
+						bool& hasNext);
 	std::vector<LeafQueryResult> const& GetResults() const {return m_results;}
 };
 
@@ -50,11 +51,12 @@ class SIDX_DLL LeafQueryResult
 {
 private:
     std::vector<SpatialIndex::id_type> ids;
-    std::unique_ptr< SpatialIndex::Region > bounds;
+    SpatialIndex::Region* bounds;
     SpatialIndex::id_type m_id;
+    LeafQueryResult();
 public:
-    LeafQueryResult() = delete;
-    LeafQueryResult(SpatialIndex::id_type id) : m_id(id){}
+    LeafQueryResult(SpatialIndex::id_type id) : bounds(0), m_id(id){}
+    ~LeafQueryResult() {if (bounds!=0) delete bounds;}
 
     /// Copy constructor.
     LeafQueryResult(LeafQueryResult const& other);

--- a/include/spatialindex/capi/ObjVisitor.h
+++ b/include/spatialindex/capi/ObjVisitor.h
@@ -34,18 +34,18 @@ class SIDX_DLL ObjVisitor : public SpatialIndex::IVisitor
 {
 private:
     std::vector<SpatialIndex::IData*> m_vector;
-    uint64_t nResults{0};
+    uint64_t nResults;
 
 public:
 
     ObjVisitor();
-    ~ObjVisitor() override;
+    ~ObjVisitor();
 
     uint64_t GetResultCount() const { return nResults; }
     std::vector<SpatialIndex::IData*>& GetResults()  { return m_vector; }
     
-    void visitNode(const SpatialIndex::INode& n) override;
-    void visitData(const SpatialIndex::IData& d) override;
-    void visitData(std::vector<const SpatialIndex::IData*>& v) override;
+    void visitNode(const SpatialIndex::INode& n);
+    void visitData(const SpatialIndex::IData& d);
+    void visitData(std::vector<const SpatialIndex::IData*>& v);
 };
 

--- a/include/spatialindex/capi/sidx_config.h
+++ b/include/spatialindex/capi/sidx_config.h
@@ -54,7 +54,7 @@
 
 #else
 
-   #include <cstdint>
+   #include <stdint.h>
    #define SIDX_THREAD  __thread
    #define STRDUP strdup
 #endif

--- a/src/capi/BoundsQuery.cc
+++ b/src/capi/BoundsQuery.cc
@@ -28,9 +28,9 @@
 
 #include <spatialindex/capi/sidx_impl.h>
 
-BoundsQuery::BoundsQuery()
-    : m_bounds{ new SpatialIndex::Region }
+BoundsQuery::BoundsQuery() 
 {
+	m_bounds = new SpatialIndex::Region;
 }
 
 void BoundsQuery::getNextEntry( const SpatialIndex::IEntry& entry, 

--- a/src/capi/CountVisitor.cc
+++ b/src/capi/CountVisitor.cc
@@ -28,11 +28,14 @@
 
 #include <spatialindex/capi/sidx_impl.h>
 
-CountVisitor::CountVisitor()
-= default;
+CountVisitor::CountVisitor(): nResults(0)
+{
+}
 
 CountVisitor::~CountVisitor()
-= default;
+{
+
+}
 
 void CountVisitor::visitNode(const SpatialIndex::INode& )
 {

--- a/src/capi/DataStream.cc
+++ b/src/capi/DataStream.cc
@@ -35,6 +35,8 @@ DataStream::DataStream(int (*readNext)(SpatialIndex::id_type * id,
 					   uint32_t *nDimension,
 					   const uint8_t** pData,
 					   size_t *nDataLength) )
+  : m_pNext(0),
+    m_bDoneReading(false)
 {
 	iterfunct = readNext;
 
@@ -42,16 +44,21 @@ DataStream::DataStream(int (*readNext)(SpatialIndex::id_type * id,
 	readData();
 }
 
+DataStream::~DataStream()
+{
+	if (m_pNext != 0) delete m_pNext;
+}
+
 bool DataStream::readData()
 {
 	SpatialIndex::id_type id;
-	double *pMin=nullptr;
-	double *pMax=nullptr;
+	double *pMin=0;
+	double *pMax=0;
 	uint32_t nDimension=0;
-	uint8_t *p_data=nullptr;
+	uint8_t *p_data=0;
 	size_t nDataLength=0;
 
-	if (m_bDoneReading) {
+	if (m_bDoneReading == true) {
 		return false;
 	}
 
@@ -68,7 +75,7 @@ bool DataStream::readData()
 	SpatialIndex::Region r = SpatialIndex::Region(pMin, pMax, nDimension);
 
 	// Data gets copied here anyway. We should fix this part of SpatialIndex::RTree::Data's constructor
-	m_pNext.reset( new SpatialIndex::RTree::Data(nDataLength, p_data, r, id) );
+	m_pNext = new SpatialIndex::RTree::Data(nDataLength, p_data, r, id);
 
 	return true;
 }
@@ -76,16 +83,17 @@ bool DataStream::readData()
 
 SpatialIndex::IData* DataStream::getNext()
 {
-	if (!m_pNext) return nullptr;
+	if (m_pNext == 0) return 0;
 
-	std::unique_ptr< SpatialIndex::RTree::Data > ret = std::move( m_pNext );
+	SpatialIndex::RTree::Data* ret = m_pNext;
+	m_pNext = 0;
 	readData();
-	return ret.release();
+	return ret;
 }
 
 bool DataStream::hasNext()
 {
-	return (static_cast< bool >(m_pNext));
+	return (m_pNext != 0);
 }
 
 uint32_t DataStream::size()
@@ -97,5 +105,9 @@ void DataStream::rewind()
 {
 	throw Tools::NotSupportedException("Operation not supported.");
 
-	m_pNext.reset();
+	if (m_pNext != 0)
+	{
+	 delete m_pNext;
+	 m_pNext = 0;
+	}
 }

--- a/src/capi/Error.cc
+++ b/src/capi/Error.cc
@@ -35,7 +35,12 @@ Error::Error(int code, std::string const& message, std::string const& method) :
 {
 }
 
-Error::Error(Error const& other) = default;
+Error::Error(Error const& other) :
+    m_code(other.m_code),
+    m_message(other.m_message),
+    m_method(other.m_method)
+{
+}
 
 Error& Error::operator=(Error const& rhs)
 {

--- a/src/capi/IdVisitor.cc
+++ b/src/capi/IdVisitor.cc
@@ -28,11 +28,14 @@
 
 #include <spatialindex/capi/sidx_impl.h>
 
-IdVisitor::IdVisitor()
-= default;
+IdVisitor::IdVisitor(): nResults(0)
+{
+}
 
 IdVisitor::~IdVisitor()
-= default;
+{
+
+}
 
 void IdVisitor::visitNode(const SpatialIndex::INode& )
 {

--- a/src/capi/Index.cc
+++ b/src/capi/Index.cc
@@ -32,7 +32,7 @@ SpatialIndex::ISpatialIndex* Index::CreateIndex()
 {
 	using namespace SpatialIndex;
 
-	ISpatialIndex* index = nullptr;
+	ISpatialIndex* index = 0;
 
 	Tools::Variant var;
 
@@ -192,9 +192,9 @@ Index::Index(	const Tools::PropertySet& poProperties,
 SpatialIndex::StorageManager::IBuffer* Index::CreateIndexBuffer(SpatialIndex::IStorageManager& storage)
 {
 	using namespace SpatialIndex::StorageManager;
-	IBuffer* buffer = nullptr;
+	IBuffer* buffer = 0;
 	try {
-		if ( m_storage == nullptr )
+		if ( m_storage == 0 )
 			throw std::runtime_error("Storage was invalid to create index buffer");
 		buffer = returnRandomEvictionsBuffer(storage, m_properties);
 	} catch (Tools::Exception& e) {
@@ -210,7 +210,7 @@ SpatialIndex::IStorageManager* Index::CreateStorage()
 {
 	using namespace SpatialIndex::StorageManager;
 
-	SpatialIndex::IStorageManager* storage = nullptr;
+	SpatialIndex::IStorageManager* storage = 0;
 	std::string filename("");
 
 	Tools::Variant var;
@@ -280,9 +280,9 @@ void Index::Initialize()
 void Index::Setup()
 
 {
-	m_buffer = nullptr;
-	m_storage = nullptr;
-	m_rtree = nullptr;
+	m_buffer = 0;
+	m_storage = 0;
+	m_rtree = 0;
 }
 
 RTIndexType Index::GetIndexType()

--- a/src/capi/LeafQuery.cc
+++ b/src/capi/LeafQuery.cc
@@ -28,6 +28,11 @@
 
 #include <spatialindex/capi/sidx_impl.h>
 
+LeafQuery::LeafQuery() 
+{
+
+}
+
 LeafQueryResult get_results(const SpatialIndex::INode* n)
 {
 	LeafQueryResult result (n->getIdentifier());
@@ -54,7 +59,7 @@ void LeafQuery::getNextEntry(	const SpatialIndex::IEntry& entry,
 
 	const SpatialIndex::INode* n = dynamic_cast<const SpatialIndex::INode*>(&entry);
 
-	if (n != nullptr)
+	if (n != 0)
 	{
 		// traverse only index nodes at levels 2 and higher.
 		if (n->getLevel() > 0)
@@ -95,12 +100,12 @@ void LeafQueryResult::SetIDs(std::vector<SpatialIndex::id_type>& v)
 }
 const SpatialIndex::Region*  LeafQueryResult::GetBounds() const
 {
-    return bounds.get();
+    return bounds;
 }
 
 void LeafQueryResult::SetBounds(const SpatialIndex::Region*  b) 
 {
-    bounds.reset( new SpatialIndex::Region(*b) );
+    bounds = new SpatialIndex::Region(*b);
 }
 
 LeafQueryResult::LeafQueryResult(LeafQueryResult const& other)
@@ -109,7 +114,7 @@ LeafQueryResult::LeafQueryResult(LeafQueryResult const& other)
     std::copy(other.ids.begin(), other.ids.end(), ids.begin());
     m_id = other.m_id;
     
-    bounds.reset( other.bounds->clone() );
+    bounds = other.bounds->clone();
 }
 
 LeafQueryResult& LeafQueryResult::operator=(LeafQueryResult const& rhs)
@@ -119,7 +124,7 @@ LeafQueryResult& LeafQueryResult::operator=(LeafQueryResult const& rhs)
         ids.resize(rhs.ids.size());
         std::copy(rhs.ids.begin(), rhs.ids.end(), ids.begin());
         m_id = rhs.m_id;
-        bounds.reset( rhs.bounds->clone() );
+        bounds = rhs.bounds->clone();
     }
     return *this;
 }

--- a/src/capi/ObjVisitor.cc
+++ b/src/capi/ObjVisitor.cc
@@ -28,8 +28,9 @@
 
 #include <spatialindex/capi/sidx_impl.h>
 
-ObjVisitor::ObjVisitor()
-= default;
+ObjVisitor::ObjVisitor(): nResults(0)
+{
+}
 
 ObjVisitor::~ObjVisitor()
 {

--- a/src/capi/Utility.cc
+++ b/src/capi/Utility.cc
@@ -144,11 +144,11 @@ Tools::PropertySet* GetDefaults()
 
     // Custom storage manager properties
     var.m_varType = Tools::VT_ULONG;
-	var.m_val.pcVal = nullptr;
+	var.m_val.pcVal = 0;
 	ps->setProperty("CustomStorageCallbacksSize", var);
 
     var.m_varType = Tools::VT_PVOID;
-	var.m_val.pcVal = nullptr;
+	var.m_val.pcVal = 0;
 	ps->setProperty("CustomStorageCallbacks", var);
 
     return ps;

--- a/src/capi/sidx_api.cc
+++ b/src/capi/sidx_api.cc
@@ -115,7 +115,7 @@ SIDX_C_DLL char* Error_GetLastErrorMsg(void){
     if (last_error.code) {
         return STRDUP(last_error.message);
     } else {
-        return nullptr;
+        return NULL;
     }
 #else
 	if (errors.empty())
@@ -132,7 +132,7 @@ SIDX_C_DLL char* Error_GetLastErrorMethod(void){
     if (last_error.code) {
         return STRDUP(last_error.method);
     } else {
-        return nullptr;
+        return NULL;
     }
 #else
 	if (errors.empty())
@@ -178,20 +178,20 @@ SIDX_C_DLL IndexH Index_Create(IndexPropertyH hProp)
 		Error_PushError(RT_Failure,
 						e.what().c_str(),
 						"Index_Create");
-		return nullptr;
+		return NULL;
 	} catch (std::exception const& e)
 	{
 		Error_PushError(RT_Failure,
 						e.what(),
 						"Index_Create");
-		return nullptr;
+		return NULL;
 	} catch (...) {
 		Error_PushError(RT_Failure,
 						"Unknown Error",
 						"Index_Create");
-		return nullptr;
+		return NULL;
 	}
-	return nullptr;
+	return NULL;
 }
 
 SIDX_C_DLL IndexH Index_CreateWithStream( IndexPropertyH hProp,
@@ -214,14 +214,14 @@ SIDX_C_DLL IndexH Index_CreateWithStream( IndexPropertyH hProp,
 		Error_PushError(RT_Failure,
 						e.what(),
 						"Index_CreateWithStream");
-		return nullptr;
+		return NULL;
 	} catch (...) {
 		Error_PushError(RT_Failure,
 						"Unknown Error",
 						"Index_CreateWithStream");
-		return nullptr;
+		return NULL;
 	}
-	return nullptr;
+	return NULL;
 }
 
 SIDX_C_DLL void Index_Destroy(IndexH index)
@@ -370,7 +370,7 @@ SIDX_C_DLL RTError Index_InsertTPData( IndexH index,
   // instead of a SpatialIndex::MovingRegion
 
   bool isPoint = false;
-  SpatialIndex::IShape* shape = nullptr;
+  SpatialIndex::IShape* shape = 0;
   double const epsilon = std::numeric_limits<double>::epsilon();
 
   double length(0), vlength(0);
@@ -443,7 +443,7 @@ SIDX_C_DLL RTError Index_InsertMVRData( IndexH index,
   // instead of a SpatialIndex::timeRegion
 
   bool isPoint = false;
-  SpatialIndex::IShape* shape = nullptr;
+  SpatialIndex::IShape* shape = 0;
   double const epsilon = std::numeric_limits<double>::epsilon();
 
   double length(0);
@@ -510,7 +510,7 @@ SIDX_C_DLL RTError Index_InsertData(  IndexH index,
 	// instead of a SpatialIndex::Region
 
 	bool isPoint = false;
-	SpatialIndex::IShape* shape = nullptr;
+	SpatialIndex::IShape* shape = 0;
 	double const epsilon = std::numeric_limits<double>::epsilon();
 
 	double length(0);
@@ -573,7 +573,7 @@ SIDX_C_DLL RTError Index_TPIntersects_obj(  IndexH index,
   VALIDATE_POINTER1(index, "Index_TPIntersects_obj", RT_Failure);
   Index* idx = reinterpret_cast<Index*>(index);
   int64_t nResultLimit, nStart;
-  SpatialIndex::MovingRegion* r = nullptr;
+  SpatialIndex::MovingRegion* r = 0;
   
   nResultLimit = idx->GetResultSetLimit();
   nStart = idx->GetResultSetOffset();
@@ -628,7 +628,7 @@ SIDX_C_DLL RTError Index_MVRIntersects_obj(  IndexH index,
   VALIDATE_POINTER1(index, "Index_MVRIntersects_obj", RT_Failure);
   Index* idx = reinterpret_cast<Index*>(index);
   int64_t nResultLimit, nStart;
-  SpatialIndex::TimeRegion* r = nullptr;
+  SpatialIndex::TimeRegion* r = 0;
 
   nResultLimit = idx->GetResultSetLimit();
   nStart = idx->GetResultSetOffset();
@@ -686,7 +686,7 @@ SIDX_C_DLL RTError Index_Intersects_obj(  IndexH index,
 	nStart = idx->GetResultSetOffset();
 
 	ObjVisitor* visitor = new ObjVisitor;
-  SpatialIndex::Region* r = nullptr;
+  SpatialIndex::Region* r = 0;
 	try {
     r = new SpatialIndex::Region(pdMin, pdMax, nDimension);
 		idx->index().intersectsWithQuery(	*r,
@@ -737,7 +737,7 @@ SIDX_C_DLL RTError Index_TPIntersects_id(  IndexH index,
   Index* idx = reinterpret_cast<Index*>(index);
 
   int64_t nResultLimit, nStart;
-  SpatialIndex::MovingRegion* r = nullptr;
+  SpatialIndex::MovingRegion* r = 0;
 
   nResultLimit = idx->GetResultSetLimit();
   nStart = idx->GetResultSetOffset();
@@ -792,7 +792,7 @@ SIDX_C_DLL RTError Index_MVRIntersects_id(  IndexH index,
   Index* idx = reinterpret_cast<Index*>(index);
 
   int64_t nResultLimit, nStart;
-  SpatialIndex::TimeRegion* r = nullptr;
+  SpatialIndex::TimeRegion* r = 0;
 
   nResultLimit = idx->GetResultSetLimit();
   nStart = idx->GetResultSetOffset();
@@ -845,7 +845,7 @@ SIDX_C_DLL RTError Index_Intersects_id(	  IndexH index,
 	Index* idx = reinterpret_cast<Index*>(index);
 
 	int64_t nResultLimit, nStart;
-  SpatialIndex::Region* r = nullptr;
+  SpatialIndex::Region* r = 0;
 
 	nResultLimit = idx->GetResultSetLimit();
   nStart = idx->GetResultSetOffset();
@@ -900,7 +900,7 @@ SIDX_C_DLL RTError Index_TPIntersects_count(	  IndexH index,
 {
   VALIDATE_POINTER1(index, "Index_TPIntersects_count", RT_Failure);
   Index* idx = reinterpret_cast<Index*>(index);
-  SpatialIndex::MovingRegion* r = nullptr;
+  SpatialIndex::MovingRegion* r = 0;
 
   CountVisitor* visitor = new CountVisitor;
   try {
@@ -949,7 +949,7 @@ SIDX_C_DLL RTError Index_MVRIntersects_count(	  IndexH index,
 {
   VALIDATE_POINTER1(index, "Index_MVRIntersects_count", RT_Failure);
   Index* idx = reinterpret_cast<Index*>(index);
-  SpatialIndex::TimeRegion* r = nullptr;
+  SpatialIndex::TimeRegion* r = 0;
 
   CountVisitor* visitor = new CountVisitor;
   try {
@@ -997,7 +997,7 @@ SIDX_C_DLL RTError Index_Intersects_count(	  IndexH index,
 {
 	VALIDATE_POINTER1(index, "Index_Intersects_count", RT_Failure);
 	Index* idx = reinterpret_cast<Index*>(index);
-  SpatialIndex::Region* r = nullptr;
+  SpatialIndex::Region* r = 0;
 
 	CountVisitor* visitor = new CountVisitor;
 	try {
@@ -1048,7 +1048,7 @@ SIDX_C_DLL RTError Index_SegmentIntersects_obj(  IndexH index,
 	Index* idx = reinterpret_cast<Index*>(index);
 
   int64_t nResultLimit, nStart;
-  SpatialIndex::LineSegment* l = nullptr;
+  SpatialIndex::LineSegment* l = 0;
 
   nResultLimit = idx->GetResultSetLimit();
   nStart = idx->GetResultSetOffset();
@@ -1101,7 +1101,7 @@ SIDX_C_DLL RTError Index_SegmentIntersects_id(	  IndexH index,
 	VALIDATE_POINTER1(index, "Index_Intersects_id", RT_Failure);
 	Index* idx = reinterpret_cast<Index*>(index);
   int64_t nResultLimit, nStart;
-  SpatialIndex::LineSegment* l = nullptr;
+  SpatialIndex::LineSegment* l = 0;
 
   nResultLimit = idx->GetResultSetLimit();
   nStart = idx->GetResultSetOffset();
@@ -1152,7 +1152,7 @@ SIDX_C_DLL RTError Index_SegmentIntersects_count(	  IndexH index,
 {
 	VALIDATE_POINTER1(index, "Index_Intersects_count", RT_Failure);
 	Index* idx = reinterpret_cast<Index*>(index);
-  SpatialIndex::LineSegment* l = nullptr;
+  SpatialIndex::LineSegment* l = 0;
 
 	CountVisitor* visitor = new CountVisitor;
 	try {
@@ -1206,7 +1206,7 @@ SIDX_C_DLL RTError Index_TPNearestNeighbors_id(IndexH index,
   VALIDATE_POINTER1(index, "Index_TPNearestNeighbors_id", RT_Failure);
   Index* idx = reinterpret_cast<Index*>(index);
   int64_t nResultLimit, nStart;
-  SpatialIndex::MovingRegion* r = nullptr;
+  SpatialIndex::MovingRegion* r = 0;
 
   nResultLimit = idx->GetResultSetLimit();
   nStart = idx->GetResultSetOffset();
@@ -1263,7 +1263,7 @@ SIDX_C_DLL RTError Index_MVRNearestNeighbors_id(IndexH index,
   VALIDATE_POINTER1(index, "Index_MVRNearestNeighbors_id", RT_Failure);
   Index* idx = reinterpret_cast<Index*>(index);
   int64_t nResultLimit, nStart;
-  SpatialIndex::TimeRegion* r = nullptr;
+  SpatialIndex::TimeRegion* r = 0;
 
   nResultLimit = idx->GetResultSetLimit();
   nStart = idx->GetResultSetOffset();
@@ -1318,7 +1318,7 @@ SIDX_C_DLL RTError Index_NearestNeighbors_id(IndexH index,
 	VALIDATE_POINTER1(index, "Index_NearestNeighbors_id", RT_Failure);
 	Index* idx = reinterpret_cast<Index*>(index);
   int64_t nResultLimit, nStart;
-  SpatialIndex::Region* r = nullptr;
+  SpatialIndex::Region* r = 0;
 
   nResultLimit = idx->GetResultSetLimit();
   nStart = idx->GetResultSetOffset();
@@ -1379,7 +1379,7 @@ SIDX_C_DLL RTError Index_TPNearestNeighbors_obj(IndexH index,
   Index* idx = reinterpret_cast<Index*>(index);
 
   int64_t nResultLimit, nStart;
-  SpatialIndex::MovingRegion* r = nullptr;
+  SpatialIndex::MovingRegion* r = 0;
 
   nResultLimit = idx->GetResultSetLimit();
   nStart = idx->GetResultSetOffset();
@@ -1436,7 +1436,7 @@ SIDX_C_DLL RTError Index_MVRNearestNeighbors_obj(IndexH index,
   Index* idx = reinterpret_cast<Index*>(index);
 
   int64_t nResultLimit, nStart;
-  SpatialIndex::TimeRegion* r = nullptr;
+  SpatialIndex::TimeRegion* r = 0;
 
   nResultLimit = idx->GetResultSetLimit();
   nStart = idx->GetResultSetOffset();
@@ -1492,7 +1492,7 @@ SIDX_C_DLL RTError Index_NearestNeighbors_obj(IndexH index,
 	Index* idx = reinterpret_cast<Index*>(index);
 
   int64_t nResultLimit, nStart;
-  SpatialIndex::Region* r = nullptr;
+  SpatialIndex::Region* r = 0;
 
   nResultLimit = idx->GetResultSetLimit();
   nStart = idx->GetResultSetOffset();
@@ -1551,7 +1551,7 @@ SIDX_C_DLL RTError Index_GetBounds(	  IndexH index,
 		idx->index().queryStrategy( *query);
 
 		const SpatialIndex::Region* bounds = query->GetBounds();
-		if (bounds == nullptr) {
+		if (bounds == 0) {
 			*nDimension = 0;
 			delete query;
 			return RT_None;
@@ -1673,9 +1673,9 @@ SIDX_C_DLL void Index_DestroyObjResults(IndexItemH* results, uint32_t nResults)
 	VALIDATE_POINTER0(results, "Index_DestroyObjResults");
 	SpatialIndex::IData* it;
 	for (uint32_t i=0; i< nResults; ++i) {
-		if (results[i] != nullptr) {
+		if (results[i] != NULL) {
 			it = reinterpret_cast<SpatialIndex::IData*>(results[i]);
-			if (it != nullptr)
+			if (it != 0)
 				delete it;
 		}
 	}
@@ -1687,7 +1687,7 @@ SIDX_C_DLL void Index_DestroyObjResults(IndexItemH* results, uint32_t nResults)
 SIDX_C_DLL void Index_Free(void* results)
 {
 	VALIDATE_POINTER0(results, "Index_Free");
-	if (results != nullptr)
+	if (results != 0)
 	    std::free(results);
 }
 
@@ -1704,7 +1704,7 @@ SIDX_C_DLL RTError Index_GetLeaves(	IndexH index,
 	Index* idx = reinterpret_cast<Index*>(index);
 
 	std::vector<LeafQueryResult>::const_iterator i;
-	LeafQuery* query = nullptr; 
+	LeafQuery* query = 0; 
 
 	// Fetch the dimensionality of the index
 	Tools::PropertySet ps;
@@ -1795,7 +1795,7 @@ SIDX_C_DLL void IndexItem_Destroy(IndexItemH item)
 {
 	VALIDATE_POINTER0(item, "IndexItem_Destroy");
 	SpatialIndex::IData* it = reinterpret_cast<SpatialIndex::IData*>(item);
-	if (it != nullptr) delete it;
+	if (it != 0) delete it;
 }
 
 SIDX_C_DLL RTError IndexItem_GetData( IndexItemH item,
@@ -1840,7 +1840,7 @@ SIDX_C_DLL RTError IndexItem_GetBounds(	  IndexItemH item,
 	SpatialIndex::Region *bounds = new SpatialIndex::Region();
     s->getMBR(*bounds);
 
-	if (bounds == nullptr) {
+	if (bounds == 0) {
 		*nDimension = 0;
                 delete bounds;
                 delete s;
@@ -1851,7 +1851,7 @@ SIDX_C_DLL RTError IndexItem_GetBounds(	  IndexItemH item,
 	*ppdMin = (double*) malloc (*nDimension * sizeof(double));
 	*ppdMax = (double*) malloc (*nDimension * sizeof(double));
 
-	if (ppdMin == nullptr || ppdMax == nullptr) {
+	if (ppdMin == NULL || ppdMax == NULL) {
 		Error_PushError(RT_Failure,
 						"Unable to allocation bounds array(s)",
 						"IndexItem_GetBounds");
@@ -1879,7 +1879,7 @@ SIDX_C_DLL void IndexProperty_Destroy(IndexPropertyH hProp)
 {
 	VALIDATE_POINTER0(hProp, "IndexProperty_Destroy");
 	Tools::PropertySet* prop = reinterpret_cast<Tools::PropertySet*>(hProp);
-	if (prop != nullptr) delete prop;
+	if (prop != 0) delete prop;
 }
 
 SIDX_C_DLL RTError IndexProperty_SetIndexType(IndexPropertyH hProp,
@@ -3198,7 +3198,7 @@ SIDX_C_DLL char* IndexProperty_GetFileName(IndexPropertyH hProp)
 			Error_PushError(RT_Failure,
 							"Property FileName must be Tools::VT_PCHAR",
 							"IndexProperty_GetFileName");
-			return nullptr;
+			return NULL;
 		}
 
 		return STRDUP(var.m_val.pcVal);
@@ -3208,7 +3208,7 @@ SIDX_C_DLL char* IndexProperty_GetFileName(IndexPropertyH hProp)
 	Error_PushError(RT_Failure,
 					"Property FileName was empty",
 					"IndexProperty_GetFileName");
-	return nullptr;
+	return NULL;
 }
 
 
@@ -3262,7 +3262,7 @@ SIDX_C_DLL char* IndexProperty_GetFileNameExtensionDat(IndexPropertyH hProp)
 			Error_PushError(RT_Failure,
 							"Property FileNameDat must be Tools::VT_PCHAR",
 							"IndexProperty_GetFileNameExtensionDat");
-			return nullptr;
+			return NULL;
 		}
 
 		return STRDUP(var.m_val.pcVal);
@@ -3272,7 +3272,7 @@ SIDX_C_DLL char* IndexProperty_GetFileNameExtensionDat(IndexPropertyH hProp)
 	Error_PushError(RT_Failure,
 					"Property FileNameDat was empty",
 					"IndexProperty_GetFileNameExtensionDat");
-	return nullptr;
+	return NULL;
 }
 
 SIDX_C_DLL RTError IndexProperty_SetFileNameExtensionIdx( IndexPropertyH hProp,
@@ -3325,7 +3325,7 @@ SIDX_C_DLL char* IndexProperty_GetFileNameExtensionIdx(IndexPropertyH hProp)
 			Error_PushError(RT_Failure,
 							"Property FileNameIdx must be Tools::VT_PCHAR",
 							"IndexProperty_GetFileNameExtensionIdx");
-			return nullptr;
+			return NULL;
 		}
 
 		return STRDUP(var.m_val.pcVal);
@@ -3335,7 +3335,7 @@ SIDX_C_DLL char* IndexProperty_GetFileNameExtensionIdx(IndexPropertyH hProp)
 	Error_PushError(RT_Failure,
 					"Property FileNameIdx was empty",
 					"IndexProperty_GetFileNameExtensionIdx");
-	return nullptr;
+	return NULL;
 }
 
 SIDX_C_DLL RTError IndexProperty_SetCustomStorageCallbacksSize(IndexPropertyH hProp,
@@ -3429,7 +3429,7 @@ SIDX_C_DLL RTError IndexProperty_SetCustomStorageCallbacks( IndexPropertyH hProp
                             new SpatialIndex::StorageManager::CustomStorageManagerCallbacks(
                                     *static_cast<const SpatialIndex::StorageManager::CustomStorageManagerCallbacks*>(value)
                                     )
-                            : nullptr;
+                            : 0;
 		prop->setProperty("CustomStorageCallbacks", var);
 
 	} catch (Tools::Exception& e)
@@ -3467,7 +3467,7 @@ SIDX_C_DLL void* IndexProperty_GetCustomStorageCallbacks(IndexPropertyH hProp)
 			Error_PushError(RT_Failure,
 							"Property CustomStorageCallbacks must be Tools::VT_PVOID",
 							"IndexProperty_GetCustomStorageCallbacks");
-			return nullptr;
+			return NULL;
 		}
 
 		return var.m_val.pvVal;
@@ -3477,7 +3477,7 @@ SIDX_C_DLL void* IndexProperty_GetCustomStorageCallbacks(IndexPropertyH hProp)
 	Error_PushError(RT_Failure,
 					"Property CustomStorageCallbacks was empty",
 					"IndexProperty_GetCustomStorageCallbacks");
-	return nullptr;
+	return NULL;
 }
 
 SIDX_C_DLL RTError IndexProperty_SetIndexID(IndexPropertyH hProp,

